src/pages/dogs/[id].jsx

```diff
@@ -13,175 +19,318 @@ import Image from "next/image";
 import maleicon from "../../../public/maleicon.svg";
 import femaleicon from "../../../public/femaleicon.svg";
 import dogplaceholdericon from "../../../public/dogplaceholdericon.svg";
+import FormField from "@/components/FormField";
+import { dogSchema } from "@/utils/consts";
+import { Nothing_You_Could_Do } from "next/font/google";
 /**
  *
  * @returns {React.ReactElement} The individual Dog page
  */
 export default function IndividualDogPage() {
   const [data, setData] = useState();
+  const [isEdit, setIsEdit] = useState(false);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm({
+    resolver: zodResolver(dogSchema.partial()),
+    defaultValues: computeDefaultValues(null),
+  });
 
   const router = useRouter();
 
   useEffect(() => {
     if (router.query.id) {
       fetch(`/api/dogs/${router.query.id}`)
         .then((res) => res.json())
-        .then((data) => setData(data));
+        .then((data) => {
+          setData(data);
+          reset(computeDefaultValues(data.data));
+        });
     }
-  }, [router.query]);
+  }, [router.query, reset]);
 
   if (!data || data === undefined || !data.success) {
     return <div>loading</div>;
   }
 
   const dog = data.data;
 
-  const dogInformationSchema = {
-    ["Birth"]: {
-      ["Birth Time"]: "N/A",
-      ["Color Color"]: "N/A",
-      ["Supplemental Feeding"]: "N/A",
-      ["Delivery Information"]: "N/A",
-      ["Birth Order"]: "N/A",
-    },
-    ["Family"]: {
-      ["Litter Size"]: "N/A",
-      ["Litter Composition"]: "N/A",
-      ["Father"]: "N/A",
-      ["Mother"]: "N/A",
-    },
-    ["Maternal Demeanor"]: {
-      ["Prior to Whelping"]: "N/A",
-      ["During Whelping"]: "N/A",
-      ["Subsequent to Whelping"]: "N/A",
-    },
-    ["Housing"]: {
-      ["Housing"]: "N/A",
-      ["Instructor"]: "N/A",
-      ["Primary Caregiver(s)"]: "N/A",
-      ["Primary Toileting Area"]: "N/A",
-    },
-    ["Feeding"]: {
-      ["Amount"]: "N/A",
-      ["First Meal"]: "N/A",
-      ["Second Meal"]: "N/A",
-      ["Third Meal"]: "N/A",
-    },
-    ["Grooming"]: {
-      ["Last bath"]: "N/A",
-    },
+  const notify = (message) => {
+    if (message === "success") {
+      toast.success("Successfully updated!");
+    } else if (message === "failure") {
+      toast.error("Unable to update!");
+    }
+  };
+
+  const onSubmit = async (data) => {
```

Comment:
rename this function to `onEditSubmit()` since there will be a few forms on this page